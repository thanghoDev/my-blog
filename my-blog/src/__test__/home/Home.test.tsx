import React from 'react';
import { render, screen } from '@testing-library/react';

import Home, { getStaticProps } from '@/pages/index';

describe('render home page', () => {
  const health = [
    {
      id: 1,
      title: 'title health',
      description: 'health description',
      category: 'health',
      view: 111,
      alt: 'alt health',
      images: 'https://i.ibb.co/10hj3h1/images.jpg',
      content: 'content',
    },
    {
      id: 2,
      title: 'title health2',
      description: 'description health2',
      category: 'health',
      view: 111,
      alt: 'alt health2',
      images: 'https://i.ibb.co/10hj3h1/images.jpg',
      content: 'content health2',
    },
  ];
  const trending = [
    {
      id: 1,
      title: 'title trending',
      description: 'description trending',
      category: 'trending',
      view: 111,
      alt: 'alt trending',
      images: 'https://i.ibb.co/10hj3h1/images.jpg',
      content: 'content trending',
    },
    {
      id: 2,
      title: 'title trending2',
      description: 'description trending2',
      category: 'category',
      view: 111,
      alt: 'alt trending2',
      images: 'https://i.ibb.co/10hj3h1/images.jpg',
      content: 'content trending2',
    },
  ];
  const business = [
    {
      id: 1,
      title: 'title business',
      description: 'description business',
      category: 'business',
      view: 111,
      alt: 'alt business',
      images: 'https://i.ibb.co/10hj3h1/images.jpg',
      content: 'content business',
    },
    {
      id: 2,
      title: 'title business2',
      description: 'description business2',
      category: 'business',
      view: 111,
      alt: 'alt business2',
      images: 'https://i.ibb.co/10hj3h1/images.jpg',
      content: 'content business2',
    },
  ];
  const politics = [
    {
      id: 1,
      title: 'title politics',
      description: 'description politics',
      category: 'politics',
      view: 111,
      alt: 'alt politics',
      images: 'https://i.ibb.co/10hj3h1/images.jpg',
      content: 'content politics',
    },
    {
      id: 2,
      title: 'title politics2',
      description: 'description politics2',
      category: 'politics',
      view: 111,
      alt: 'alt politics2',
      images: 'https://i.ibb.co/10hj3h1/images.jpg',
      content: 'content politics2',
    },
  ];
  const carousel = [
    {
      id: 1,
      title: 'title carousel',
      description: 'description carousel',
      category: 'carousel',
      view: 111,
      alt: 'alt carousel',
      images: 'https://i.ibb.co/10hj3h1/images.jpg',
      content: 'content carousel',
    },
    {
      id: 2,
      title: 'title carousel2',
      description: 'description carousel2',
      category: 'carousel',
      view: 111,
      alt: 'alt carousel2',
      images: 'https://i.ibb.co/10hj3h1/images.jpg',
      content: 'content carousel2',
    },
  ];
  test('test getStaticProps', async () => {
    global.fetch = jest
      .fn()
      .mockImplementationOnce(() =>
        Promise.resolve({
          ok: true,
          json: () => carousel,
        })
      )
      .mockImplementationOnce(() =>
        Promise.resolve({
          ok: true,
          json: () => health,
        })
      )
      .mockImplementationOnce(() =>
        Promise.resolve({
          ok: true,
          json: () => trending,
        })
      )
      .mockImplementationOnce(() =>
        Promise.resolve({
          ok: true,
          json: () => business,
        })
      )
      .mockImplementationOnce(() =>
        Promise.resolve({
          ok: true,
          json: () => politics,
        })
      );

    const response = await getStaticProps();
    expect(response).toEqual({
      props: {
        carousel: carousel,
        health: health,
        trending: trending,
        business: business,
        politics: politics,
      },
      revalidate: 300,
    });
    expect(fetch).toHaveBeenCalled();
  });
  test('testing home page', () => {
    const { container } = render(
      <Home
        health={health}
        trending={trending}
        business={business}
        politics={politics}
        carousel={carousel}
      />
    );
    expect(container).toMatchSnapshot();
    health.forEach((item) => {
      expect(screen.getByText(item.title)).toBeInTheDocument();
      expect(screen.getByRole('img', { name: item.alt })).toBeInTheDocument();
    });
    trending.forEach((item) => {
      expect(screen.getByText(item.title)).toBeInTheDocument();
    });
    business.forEach((item) => {
      expect(screen.getByText(item.title)).toBeInTheDocument();
      expect(screen.getByRole('img', { name: item.alt })).toBeInTheDocument();
      expect(screen.getByText(item.description)).toBeInTheDocument();
    });
    politics.forEach((item) => {
      expect(screen.getByText(item.title)).toBeInTheDocument();
      expect(screen.getByRole('img', { name: item.alt })).toBeInTheDocument();
      expect(screen.getByText(item.description)).toBeInTheDocument();
    });
    carousel.forEach((item) => {
      expect(screen.getByText(item.title)).toBeInTheDocument();
      expect(screen.getByRole('img', { name: item.alt })).toBeInTheDocument();
    });
  });
});
