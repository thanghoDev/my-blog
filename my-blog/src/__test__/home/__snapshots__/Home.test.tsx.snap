// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render home page testing home page 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="mt-5 d-flex bg-light carousel slide carousel-dark"
    >
      <div
        class="carousel-indicators"
      >
        <button
          aria-current="true"
          aria-label="Slide 1"
          class="active"
          data-bs-target=""
          type="button"
        />
        <button
          aria-current="false"
          aria-label="Slide 2"
          data-bs-target=""
          type="button"
        />
      </div>
      <div
        class="carousel-inner"
      >
        <div
          class="active carousel-item"
        >
          <img
            alt="alt carousel"
            class="d-block"
            data-nimg="future"
            decoding="async"
            height="377"
            loading="lazy"
            src="/_next/image?url=https%3A%2F%2Fi.ibb.co%2F10hj3h1%2Fimages.jpg&w=1200&q=75"
            srcset="/_next/image?url=https%3A%2F%2Fi.ibb.co%2F10hj3h1%2Fimages.jpg&w=640&q=75 1x, /_next/image?url=https%3A%2F%2Fi.ibb.co%2F10hj3h1%2Fimages.jpg&w=1200&q=75 2x"
            style="color: transparent;"
            width="555"
          />
          <div
            class="text-dark text-start w-50 top-0 start-50 ps-4 carousel-caption"
          >
            <h3
              class="text-capitalize text-black postTitle"
            >
              <a>
                title carousel
              </a>
            </h3>
            <p
              class="description"
            >
              description carousel
            </p>
            <p
              class="text-capitalize category mt-3"
            >
              Dave Rogers 
              <code
                class="text-secondary"
              >
                in
              </code>
               
              <a
                class="text-black text-decoration-none"
                href="/category/carousel"
              >
                carousel
              </a>
            </p>
          </div>
        </div>
        <div
          class="carousel-item"
        >
          <img
            alt="alt carousel2"
            class="d-block"
            data-nimg="future"
            decoding="async"
            height="377"
            loading="lazy"
            src="/_next/image?url=https%3A%2F%2Fi.ibb.co%2F10hj3h1%2Fimages.jpg&w=1200&q=75"
            srcset="/_next/image?url=https%3A%2F%2Fi.ibb.co%2F10hj3h1%2Fimages.jpg&w=640&q=75 1x, /_next/image?url=https%3A%2F%2Fi.ibb.co%2F10hj3h1%2Fimages.jpg&w=1200&q=75 2x"
            style="color: transparent;"
            width="555"
          />
          <div
            class="text-dark text-start w-50 top-0 start-50 ps-4 carousel-caption"
          >
            <h3
              class="text-capitalize text-black postTitle"
            >
              <a>
                title carousel2
              </a>
            </h3>
            <p
              class="description"
            >
              description carousel2
            </p>
            <p
              class="text-capitalize category mt-3"
            >
              Dave Rogers 
              <code
                class="text-secondary"
              >
                in
              </code>
               
              <a
                class="text-black text-decoration-none"
                href="/category/carousel"
              >
                carousel
              </a>
            </p>
          </div>
        </div>
      </div>
      <a
        class="carousel-control-prev"
        href="#"
        role="button"
        tabindex="0"
      >
        <span
          aria-hidden="true"
          class="carousel-control-prev-icon"
        />
        <span
          class="visually-hidden"
        >
          Previous
        </span>
      </a>
      <a
        class="carousel-control-next"
        href="#"
        role="button"
        tabindex="0"
      >
        <span
          aria-hidden="true"
          class="carousel-control-next-icon"
        />
        <span
          class="visually-hidden"
        >
          Next
        </span>
      </a>
    </div>
    <div
      class="d-flex pt-5 pb-5"
    >
      <div
        class="mt-5 align-content-start row row-cols-md-8"
      >
        <h2
          class="mb-5 postTitle text-decoration-underline text-capitalize fs-5"
        >
          <a>
            health
          </a>
        </h2>
        <div
          class="col-md-6"
        >
          <img
            alt="alt health"
            data-nimg="future"
            decoding="async"
            height="210"
            loading="lazy"
            src="/_next/image?url=https%3A%2F%2Fi.ibb.co%2F10hj3h1%2Fimages.jpg&w=750&q=75"
            srcset="/_next/image?url=https%3A%2F%2Fi.ibb.co%2F10hj3h1%2Fimages.jpg&w=384&q=75 1x, /_next/image?url=https%3A%2F%2Fi.ibb.co%2F10hj3h1%2Fimages.jpg&w=750&q=75 2x"
            style="color: transparent;"
            width="350"
          />
          <div
            class="mt-5"
          >
            <a
              class="fs-5 text-black text-decoration-none text-capitalize"
              href="/detail/1"
            >
              title health
            </a>
            <p
              class="description"
            >
              health description
            </p>
          </div>
        </div>
        <div
          class="col-md-6"
        >
          <div
            class="d-flex mb-2"
          >
            <img
              alt="alt health2"
              data-nimg="future"
              decoding="async"
              height="107"
              loading="lazy"
              src="/_next/image?url=https%3A%2F%2Fi.ibb.co%2F10hj3h1%2Fimages.jpg&w=256&q=75"
              srcset="/_next/image?url=https%3A%2F%2Fi.ibb.co%2F10hj3h1%2Fimages.jpg&w=128&q=75 1x, /_next/image?url=https%3A%2F%2Fi.ibb.co%2F10hj3h1%2Fimages.jpg&w=256&q=75 2x"
              style="color: transparent;"
              width="105"
            />
            <div
              class="ms-4"
            >
              <a
                class="fs-6 text-black text-decoration-none text-capitalize"
                href="/detail/2"
              >
                title health2
              </a>
              <p
                class="text-capitalize category mt-3"
              >
                Dave Rogers 
                <code
                  class="text-secondary"
                >
                  in
                </code>
                 
                <a
                  class="text-black text-decoration-none"
                  href="/category/health"
                >
                  health
                </a>
              </p>
            </div>
          </div>
        </div>
      </div>
      <div
        class="mt-5 ps-4 col-md-4"
      >
        <h2
          class="mb-5 text-decoration-underline text-capitalize fs-5"
        >
          trending
        </h2>
        <div
          class="mt-3 d-flex"
        >
          <h3>
            0
            1
          </h3>
          <div
            class="ms-4"
          >
            <a
              class="text-black text-capitalize text-decoration-none undefined"
              href="/detail/1"
            >
              title trending
            </a>
            <p
              class="text-capitalize category mt-3"
            >
              Dave Rogers 
              <code
                class="text-secondary"
              >
                in
              </code>
               
              <a
                class="text-black text-decoration-none"
                href="/category/trending"
              >
                trending
              </a>
            </p>
          </div>
        </div>
        <div
          class="mt-3 d-flex"
        >
          <h3>
            0
            2
          </h3>
          <div
            class="ms-4"
          >
            <a
              class="text-black text-capitalize text-decoration-none undefined"
              href="/detail/2"
            >
              title trending2
            </a>
            <p
              class="text-capitalize category mt-3"
            >
              Dave Rogers 
              <code
                class="text-secondary"
              >
                in
              </code>
               
              <a
                class="text-black text-decoration-none"
                href="/category/category"
              >
                category
              </a>
            </p>
          </div>
        </div>
        <a
          class="text-uppercase link-success text-decoration-none"
          href="/#"
        >
          See all trends
          <i
            class="bi bi-chevron-right"
          />
        </a>
      </div>
    </div>
    <div
      class="d-flex pt-5 pb-5"
    >
      <div
        class="me-4 col-md-6"
      >
        <h2
          class="mb-5 text-decoration-underline text-capitalize fs-5 postTitle"
        >
          <a>
            politics
          </a>
        </h2>
        <div
          class="d-flex mb-3"
        >
          <img
            alt="alt politics"
            data-nimg="future"
            decoding="async"
            height="180"
            loading="lazy"
            src="/_next/image?url=https%3A%2F%2Fi.ibb.co%2F10hj3h1%2Fimages.jpg&w=384&q=75"
            srcset="/_next/image?url=https%3A%2F%2Fi.ibb.co%2F10hj3h1%2Fimages.jpg&w=256&q=75 1x, /_next/image?url=https%3A%2F%2Fi.ibb.co%2F10hj3h1%2Fimages.jpg&w=384&q=75 2x"
            style="color: transparent;"
            width="165"
          />
          <div
            class="ms-4"
          >
            <a
              class="fs-6 text-decoration-none text-black text-capitalize postTitle"
              href="/detail/1"
            >
              title politics
            </a>
            <p
              class="description"
            >
              description politics
            </p>
            <p
              class="text-capitalize category"
            >
              Dave Rogers 
              <code
                class="text-secondary"
              >
                in
              </code>
               
              <a
                class="text-black text-decoration-none"
                href="/category/politics"
              >
                politics
              </a>
            </p>
          </div>
        </div>
        <div
          class="d-flex mb-3"
        >
          <img
            alt="alt politics2"
            data-nimg="future"
            decoding="async"
            height="180"
            loading="lazy"
            src="/_next/image?url=https%3A%2F%2Fi.ibb.co%2F10hj3h1%2Fimages.jpg&w=384&q=75"
            srcset="/_next/image?url=https%3A%2F%2Fi.ibb.co%2F10hj3h1%2Fimages.jpg&w=256&q=75 1x, /_next/image?url=https%3A%2F%2Fi.ibb.co%2F10hj3h1%2Fimages.jpg&w=384&q=75 2x"
            style="color: transparent;"
            width="165"
          />
          <div
            class="ms-4"
          >
            <a
              class="fs-6 text-decoration-none text-black text-capitalize postTitle"
              href="/detail/2"
            >
              title politics2
            </a>
            <p
              class="description"
            >
              description politics2
            </p>
            <p
              class="text-capitalize category"
            >
              Dave Rogers 
              <code
                class="text-secondary"
              >
                in
              </code>
               
              <a
                class="text-black text-decoration-none"
                href="/category/politics"
              >
                politics
              </a>
            </p>
          </div>
        </div>
      </div>
      <div
        class="me-4 col-md-6"
      >
        <h2
          class="mb-5 text-decoration-underline text-capitalize fs-5 postTitle"
        >
          <a>
            business
          </a>
        </h2>
        <div
          class="d-flex mb-3"
        >
          <img
            alt="alt business"
            data-nimg="future"
            decoding="async"
            height="180"
            loading="lazy"
            src="/_next/image?url=https%3A%2F%2Fi.ibb.co%2F10hj3h1%2Fimages.jpg&w=384&q=75"
            srcset="/_next/image?url=https%3A%2F%2Fi.ibb.co%2F10hj3h1%2Fimages.jpg&w=256&q=75 1x, /_next/image?url=https%3A%2F%2Fi.ibb.co%2F10hj3h1%2Fimages.jpg&w=384&q=75 2x"
            style="color: transparent;"
            width="165"
          />
          <div
            class="ms-4"
          >
            <a
              class="fs-6 text-decoration-none text-black text-capitalize postTitle"
              href="/detail/1"
            >
              title business
            </a>
            <p
              class="description"
            >
              description business
            </p>
            <p
              class="text-capitalize category"
            >
              Dave Rogers 
              <code
                class="text-secondary"
              >
                in
              </code>
               
              <a
                class="text-black text-decoration-none"
                href="/category/business"
              >
                business
              </a>
            </p>
          </div>
        </div>
        <div
          class="d-flex mb-3"
        >
          <img
            alt="alt business2"
            data-nimg="future"
            decoding="async"
            height="180"
            loading="lazy"
            src="/_next/image?url=https%3A%2F%2Fi.ibb.co%2F10hj3h1%2Fimages.jpg&w=384&q=75"
            srcset="/_next/image?url=https%3A%2F%2Fi.ibb.co%2F10hj3h1%2Fimages.jpg&w=256&q=75 1x, /_next/image?url=https%3A%2F%2Fi.ibb.co%2F10hj3h1%2Fimages.jpg&w=384&q=75 2x"
            style="color: transparent;"
            width="165"
          />
          <div
            class="ms-4"
          >
            <a
              class="fs-6 text-decoration-none text-black text-capitalize postTitle"
              href="/detail/2"
            >
              title business2
            </a>
            <p
              class="description"
            >
              description business2
            </p>
            <p
              class="text-capitalize category"
            >
              Dave Rogers 
              <code
                class="text-secondary"
              >
                in
              </code>
               
              <a
                class="text-black text-decoration-none"
                href="/category/business"
              >
                business
              </a>
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="bg-light pt-5 pb-5 mt-5"
  >
    <div
      class="container pb-5 pt-5"
    >
      <form
        action="#"
        class="row align-items-center"
      >
        <div
          class="col-md-5 mr-auto"
        >
          <h2
            class="text-capitalize"
          >
            newsletter subscribe
          </h2>
          <p
            class="description"
          >
            Lorem ipsum dolor sit amet, consectetur adipisicing elit. Perferendis aspernatur ut at quae omnis pariatur obcaecati possimus nisi ea iste!
          </p>
        </div>
        <div
          class="col-md-6 ml-auto d-flex gap-2"
        >
          <input
            class="form-control rounded-pill"
            placeholder="Enter your email"
            type="email"
          />
          <button
            class="btn btn-outline-success border border-light rounded-circle bg-dark"
            type="button"
          >
            <i
              class="bi bi-send text-light"
            />
          </button>
        </div>
      </form>
    </div>
  </div>
</div>
`;
